app/modules/entities/components/cleanup/lib/get_parts_suggestions.ts

```diff
@@ -0,0 +1,59 @@
+import { uniq, property, pluck, flatten } from 'underscore'
+import { API } from '#app/api/api'
+import preq from '#app/lib/preq'
+import { getEntities, type SerializedEntity } from '#app/modules/entities/lib/entities'
+import { searchWorks } from '#entities/lib/search/search_by_types'
+import type { GetAuthorWorksResponse } from '#server/controllers/entities/get_entity_relatives'
+import type { EntityUri } from '#server/types/entity'
+import { addPertinanceScore, type WorkSuggestion } from './add_pertinance_score.ts'
+
+export async function getPartsSuggestions (serie: SerializedEntity, parts: SerializedEntity[], authorsUris: EntityUri[]) {
+  const uris = await Promise.all([
+    getAuthorsWorks(authorsUris),
+    searchMatchWorks(serie, parts),
+  ])
+  .then(flatten)
+  .then(uniq)
+
+  const works = await getEntities(uris, { refresh: true })
+
+  const partsSuggestions = works
+    // Confirm the type, as the search might have failed to unindex a serie that use
+    // to be considered a work
+    .filter(isWorkWithoutSerie)
+    .map(addPertinanceScore(serie))
+    .filter(work => work.authorMatch || work.labelMatch)
+    .sort(byDescendingPertinanceScore)
+
+  return partsSuggestions as WorkSuggestion[]
+}
+
+function byDescendingPertinanceScore (a: WorkSuggestion, b: WorkSuggestion) {
+  return b.pertinanceScore - a.pertinanceScore
+}
+
+async function getAuthorsWorks (authorsUris: EntityUri[]) {
+  const authorsWorks = await Promise.all(authorsUris.map(fetchAuthorWorks))
+  const authorsWorksUris = authorsWorks.map(works => pluck(works.filter(hasNoSerie), 'uri'))
+  return authorsWorksUris.flat()
+}
+
+async function fetchAuthorWorks (authorUri: EntityUri) {
```

Comment:
What about reusing `getAuthorWorks` which have same I/O but with some additional filters ?

Reply:
No, `getAuthorWorks`'s output is an array of serialized entities, while the one here returns only `uris`. Clarified in 687ca1642